juju/juju
# fix bug 1829393

## Description of change

The instance-mutater worker should not watch KVM containers.  Most of the change is threading the machine's ContainerType into the allwatcher, model cache, and apiserver.

## QA steps

1. juju bootstrap --config logging-config='<root>=DEBUG;unit=DEBUG;juju.worker.instancemutater=TRACE' <an open stack cloud>
3. juju deploy nova-compute --to kvm
2. juju add-unit nova-computer --to lxd:0
2. juju deploy neutron-openvswitch
3. juju add-relation nova-compute neutron-openvswitch
2. juju debug-log  --include-module juju.worker.instancemutater  

Log messages should indicated that an lxd profile was applied to 0/lxd/0 for neutron-openvswitch and that 0/kvm/0 is being ignored by the instance mutater.

Also test starting with a 2.5.7 controller and deploy units; upgrade to 2.6.2 to see the error messages from the bug; then upgrade to this code to see the error messages resolved.  

## Documentation changes

N/A

## Bug reference

https://bugs.launchpad.net/juju/+bug/1829393
